From 632cf60862dd106f89db4121b7c592e37694cd58 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Mon, 19 May 2014 01:37:46 +1000
Subject: [PATCH 1/2] Reset frames/duration maps when changing file, or when
 initialising the position map.

frame/position maps were never reset, causing the PositionMap to rarely been updated and the totalFrames and program duration to never be updated after the file had been opened.
This was responsible for wrong seeks in in-progress recordings and liveTV
---
 mythtv/libs/libmythtv/decoderbase.cpp |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mythtv/libs/libmythtv/decoderbase.cpp b/mythtv/libs/libmythtv/decoderbase.cpp
index 3277657..84868e0 100644
--- a/mythtv/libs/libmythtv/decoderbase.cpp
+++ b/mythtv/libs/libmythtv/decoderbase.cpp
@@ -197,6 +197,8 @@ bool DecoderBase::PosMapFromDb(void)
     QMutexLocker locker(&m_positionMapLock);
     m_positionMap.clear();
     m_positionMap.reserve(posMap.size());
+    m_frameToDurMap.clear();
+    m_durToFrameMap.clear();
 
     for (frm_pos_map_t::const_iterator it = posMap.begin();
          it != posMap.end(); ++it)
@@ -667,6 +669,8 @@ void DecoderBase::ResetPosMap(void)
     QMutexLocker locker(&m_positionMapLock);
     posmapStarted = false;
     m_positionMap.clear();
+    m_frameToDurMap.clear();
+    m_durToFrameMap.clear();
 }
 
 long long DecoderBase::GetLastFrameInPosMap(void) const
-- 
1.7.10.2

